src/test/java/org/zeroturnaround/zip/SkipForJava8Runner.java

```diff
@@ -0,0 +1,43 @@
+package org.zeroturnaround.zip;
+
+import java.util.zip.ZipEntry;
+
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+
+/**
+ * JUnit runner that can be used to skip tests for JDK8. This is needed
+ * for example some timestamp preserving tests that should have custom
+ * JDK 8 implementations and custom pre JDK8 tests.
+ */
+public class SkipForJava8Runner extends BlockJUnit4ClassRunner {
```

Comment:
The thing with Assumptions is that Eclipse still considers them failures when AssumptionViolations happen. They supposedly have it fixed but my Eclipse 4.7 shows assumptionviolations as failed unit tests which is not desired - hence the custom runner.